tests/Properties.hs

```diff
@@ -93,6 +96,23 @@ prop_lines_lazy2 =
 
 prop_strip x = C.strip x == (C.dropSpace . C.reverse . C.dropSpace . C.reverse) x
 
+prop_toConstr :: P.ByteString -> Property
+prop_toConstr bs = True ==> "pack" == ((showConstr  . toConstr) bs)
+
+prop_gshow_empty :: P.ByteString -> Property
+prop_gshow_empty b = (not . null . Char8.unpack) b ==> (not . null . gshow) b
+
+prop_gshow_equal :: P.ByteString -> Property
+prop_gshow_equal b = True ==> read_bs b == read_string b
+    where
+        read_bs :: P.ByteString -> [(P.ByteString, String)]
+        read_bs = gread . gshow
+        read_string :: P.ByteString -> [(P.ByteString, String)]
+        read_string = gread . Char8.unpack
+
+prop_gshow_string :: P.ByteString -> Property
+prop_gshow_string b = True ==> (gshow . Char8.pack) "A" == "(pack ((:) (65) ([])))"
```

Comment:
I was a bit surprised that the `gshow` output differs so much from the one for `text`

```
ghci> gshow @Text "A"
"(pack \"A\")"
```

But it's consistent with `gshow` for `[Word8]`, so it makes sense. :+1:

Reply:
Yeah, it's a bit weird. I've looked into how the gshow and Data instance functions work and I couldn't find a way to better define the behavior of what gshow produces. As it stands it just produces a string with all of the bytes in decimal form.